cpp/src/Ice/Reference.cpp

```diff
@@ -1128,8 +1128,8 @@ size_t
 IceInternal::RoutableReference::hash() const noexcept
 {
     size_t h = Reference::hash();
-    hashAdd(h, _endpoints);
     hashAdd(h, _adapterId);
+    hashAdd(h, _endpoints);
```

Comment:
I know it doesn't matter, but this is the order we hash them in C# and Java.
Good to be consistent I figure.